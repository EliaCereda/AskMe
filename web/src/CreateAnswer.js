import React, {Component} from "react";

class CreateAnswer extends Component {
    constructor(props) {
        super(props);

        this.state = {body: ''};

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
        const state = {};

        state[event.target.name] = event.target.value;

        this.setState(state);
    }

    handleSubmit(event) {
        this.props.onCreate(this.state);

        this.setState({body: ''});

        event.preventDefault();
    }

    render() {
        return (
            <div className="CreateAnswer">
                <form onSubmit={this.handleSubmit}>
                    <textarea name="body"
                              value={this.state.body}
                              placeholder="Write somethingâ€¦"
                              onChange={this.handleChange}
                              required/>
                    <br/>
                    <input type="submit" value="Submit"/>
                </form>
            </div>
        );
    }
}

export default CreateAnswer;
