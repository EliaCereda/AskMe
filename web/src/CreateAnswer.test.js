import React from 'react';
import CreateAnswer from "./CreateAnswer";

describe('CreateAnswer', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<CreateAnswer/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('triggers login event', () => {
        const body = "Questa Ã¨ una risposta.";

        const handleLogin = sinon.spy();
        const wrapper = shallow(<CreateAnswer onCreate={handleLogin}/>);

        wrapper.find('textarea').simulate('change', {target: {name: 'body', value: body}});
        wrapper.find('form').simulate('submit', {
            preventDefault: () => {}
        });

        expect(handleLogin.withArgs({body}).calledOnce).toBe(true);
    });
});
