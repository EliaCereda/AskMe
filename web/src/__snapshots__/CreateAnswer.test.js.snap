// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAnswer renders correctly 1`] = `
<div
  className="CreateAnswer"
>
  <form
    onSubmit={[Function]}
  >
    <textarea
      name="body"
      onChange={[Function]}
      placeholder="Write somethingâ€¦"
      required={true}
      value=""
    />
    <br />
    <input
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
